uni/android/app/build.gradle

```diff
@@ -32,16 +32,20 @@ android {
     namespace "pt.up.fe.ni.uni"
 
     compileSdkVersion 34
-    ndkVersion flutter.ndkVersion
+    ndkVersion "25.1.8937393"
 
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_17
-        targetCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_11
+        sourceCompatibility JavaVersion.VERSION_11
 
         // This is required by flutter_local_notifications. See its README.
         coreLibraryDesugaringEnabled true
     }
 
+    kotlinOptions {
+        jvmTarget = "11"
+    }
+
```

Comment:
Shouldn't this version match the one we have in CI? `21.x`

Reply:
Not really. It specifies the Java bytecode level you want to generate.

Java (and Kotlin) compiler can generate older bytecode than its version. For example, OpenJDK 21 can generate JVM bytecode from versions 8 up to 21. And OpenJDK 17 can generate JVM bytecode up to 17.

[Android doesn't actually use the JVM, but ART](https://stackoverflow.com/questions/28663243/does-the-android-system-include-a-jvm), which is an alternative implementation that's lagging behind latest Java bytecode features. In other words, if you generate too new bytecode (like 21), then it won't run on Android.

bytecode target being 11 is a safe choice. 

[Here's a great read from Jake Wharton](https://jakewharton.com/gradle-toolchains-are-rarely-a-good-idea/) explaining it in more detail.